src/charm.py

```diff
@@ -139,8 +146,14 @@ def __init__(self, *args: Any):
         self.oauth_requirer = OAuthRequirer(self, oauth_client_config, OAUTH_INTEGRATION_NAME)
         self.oauth_integration = OAuthIntegration(self.oauth_requirer)
 
-        self.certificate_transfer_requirer = CertificateTransferRequires(
-            self, CERTIFICATE_TRANSFER_INTEGRATION_NAME
+        self.certificate_transfer_requirer_v0 = CertificateTransferRequires(
+            self,
+            CERTIFICATE_TRANSFER_INTEGRATION_NAME,
+        )
+
+        self.certificate_transfer_requirer_v1 = V1CertificateTransferRequires(
+            self,
+            CERTIFICATE_TRANSFER_INTEGRATION_NAME,
         )
```

Comment:
thought: maybe we could instantiate only v1. I think that this will work as the event will still be triggered and we manually handle the parsing of the databag.